client/components/zendesk-chat-widget/index.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+
+interface Props {
+	chatId: string | false;
+}
+
+const ZendeskChat = ( { chatId }: Props ) => {
+	useEffect( () => {
+		if ( ! chatId ) {
+			return;
+		}
+
+		const script = document.createElement( 'script' );
+		script.src = 'https://static.zdassets.com/ekr/snippet.js?key=' + chatId;
+		script.type = 'text/javascript';
+		script.id = 'ze-snippet';
```

Comment:
We have a [`loadScript` utility](https://github.com/Automattic/wp-calypso/blob/trunk/packages/load-script/README.md) - would that help here?

Reply:
Unfortunately the script tag `id` needs to be specific, so it seems that won't work here

Reply:
Supporting a custom script `id` sounds like a useful addition to `loadScript`, and doesn't seem too complex to pull off, especially if done in a separate PR. Let me know if you're interested and need any help with that.

Reply:
Check the post-mortem in p58i-ch2-p2 about a signup issue that came up when Olark had a failure on their end, which caused WPCOM's signup to fail. To avoid this issue again for Zendesk, I would recommend wrapping this component in an error boundary component, like it was done [here](https://github.com/Automattic/wp-calypso/pull/62483/files#diff-65a04c39a5351c2207639b42494401dd0ce582e582332e4f2aa05ff6f321f0ddR7) and [here](https://github.com/Automattic/wp-calypso/pull/62483/files#diff-0aeb56c8f977988c48e11f68a515da183bbfc39b2af1d0a846839a6eb275b3b7R1). cc @tyxla in case you have additional thoughts on best practices here.

Reply:
I added the error boundary (btw this PR is for the signup related changes only, and is supposed to land after #70118 where I make the changes to the core Zendesk Chat component and enable it on checkout)